frontend/playground-frontend/src/components/helper_functions/TalkWithBackend.test.tsx

```diff
@@ -0,0 +1,68 @@
+import axios from "axios";
+import { render, screen, act } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { auth } from "../../firebase";
+import {
+  uploadToBackend,
+  userCodeEval,
+  getSignedUploadUrl,
+  createExecutionId,
+  sendToBackend,
+  train_and_output,
+  sendEmail,
+} from "./TalkWithBackend";
+
+jest.mock('axios', () => ({
+    post: jest.fn(() => Promise.resolve({ data: {} })),
+}));
+  
+jest.mock("../../firebase", () => ({
+  auth: {
+    currentUser: {
+      getIdToken: jest.fn(),
+      uid: "myuid",
+      email: "myemail@example.com",
+      displayName: "myname",
+    },
+  },
+}));
+
+describe("Test uploadToBackend function", () => {
+    it("should call axios.post with correct arguments", async () => {
+        const data = { key1: "value1", key2: "value2" };
+        const token = "mytoken";
+        (auth.currentUser?.getIdToken as jest.Mock).mockResolvedValueOnce(token);
```

Comment:
sometimes when it comes to firebase, we need to cast functions as a Mock so that we can control what the "resolved value" is or the implementation. weird edge case when it comes to unit testing JS/TS code with `jest`